app/controllers/partners_controller.rb

```diff
@@ -109,9 +109,13 @@ def invite
 
   def invite_partner_user
     partner = current_organization.partners.find(params[:partner])
-    UserInviteService.invite(email: params[:email],
+
+    UserInviteService.invite(
+      name: params[:name].presence || "Name Not Provided",
```

Comment:
Honestly, I don't think we should be saving "Name Not Provided" to the database. From a model perspective that doesn't make a lot of sense. Now we're complicating the workflow by having some names as blank, some as nil and some as "Name Not Provided".

The "right" approach IMO is to:

*   Change the name column to allow nulls and change the default to null
*   Add a `display_name` to the User model that returns the name or `Name Not Provided` if it's null
*   Do a mass update of all users with name `Name Not Provided`to replace with null.

@awwaiid what do you think?

Reply:
I think that's a great idea, but let's do it in a separate PR -- this one maintains the existing algorithm that writes it down in the database, so it doesn't go backwards and is otherwise complete.

I'm opening a new issue for this.

Reply:
I've changed my mind, I think we should do it in this ticket.

Reply:
AND update app/views to change `user.name` to `user.display_name`

Reply:
Nevermind, I'm coming all the way back around (while we are talking through it) into a separate PR

Reply:
@awwaiid I've resolved the branch issue and am ready for any new tasks stemming from this PR. Let's discuss potential next steps in our Sunday meeting, after a thorough team review.

Resolution Process:
I reverted the commit with the team's help via Slack, which led to losing some minor changes. This was a valuable lesson in careful version control. I'll use safer methods like git stash for temporary changes in the future to avoid similar issues.

Reply:
@awwaiid I'm a bit confused about your thoughts here... should we do it in the same PR and replace the DB saving logic? Or do it in a different PR and leave this for something else?

Reply:
@dorner Making the default name null is part of #4115

Reply:
@dorner leave it here to where the same thing ends up in the database (the default value), and then in the other PR undo the default both here and in the DB default-value column.